test/topicFilters.spec.ts

```diff
@@ -431,3 +441,171 @@ describe("Send/Receive messages using boolean filters of subscription", function
+
+    await sendOrders();
+    const receivedMsgs = await receiveOrders(subscriptionClient);
+
+    should.equal(Array.isArray(receivedMsgs), true);
+    should.equal(receivedMsgs.length, 8);
+
+    await testPeekMsgsLength(subscriptionClient, 0);
+  });
+
+  it("SQL rule filter with action", async function(): Promise<void> {
+    await subscriptionClient.addRule("SqlRuleWithAction", "color='blue'", "SET priority = 'High'");
+    const rules = await subscriptionClient.getRules();
+    should.equal(rules.length, 1);
+    should.equal(rules[0].name, "SqlRuleWithAction");
+
+    await sendOrders();
+    const receivedMsgs = await receiveOrders(subscriptionClient);
+
+    should.equal(Array.isArray(receivedMsgs), true);
+    should.equal(receivedMsgs.length, 4);
+
+    await testPeekMsgsLength(subscriptionClient, 0);
+  });
+});
+
+describe("Send/Receive messages using correlation filters of subscription", function(): void {
+  beforeEach(async () => {
+    await beforeEachTest();
+  });
+
+  afterEach(async () => {
+    await receiveOrders(defaultSubscriptionClient);
+    await testPeekMsgsLength(defaultSubscriptionClient, 0);
+
+    await afterEachTest();
+  });
+
+  it("Correlation filter on the message properties", async function(): Promise<void> {
+    await subscriptionClient.addRule("CorrelationMsgPropertyRule", {
+      label: "red",
+      correlationId: "high"
+    });
+
+    const rules = await subscriptionClient.getRules();
+    should.equal(rules.length, 1, "rule");
+    should.equal(rules[0].name, "CorrelationMsgPropertyRule");
+
+    await sendOrders();
+    const receivedMsgs = await receiveOrders(subscriptionClient);
+
+    should.equal(Array.isArray(receivedMsgs), true);
+    should.equal(receivedMsgs.length, 1);
+
+    await testPeekMsgsLength(subscriptionClient, 0);
+  });
+
+  it("Correlation filter on the custom properties", async function(): Promise<void> {
+    await subscriptionClient.addRule("CorrelationCustomRule", {
+      userProperties: {
+        color: "red"
+      }
+    });
+
+    const rules = await subscriptionClient.getRules();
+    should.equal(rules.length, 1, "rule");
+    should.equal(rules[0].name, "CorrelationCustomRule");
+
+    await sendOrders();
+    const receivedMsgs = await receiveOrders(subscriptionClient);
+
+    should.equal(Array.isArray(receivedMsgs), true);
+    should.equal(receivedMsgs.length, 4);
+
+    await testPeekMsgsLength(subscriptionClient, 0);
+  });
+
+  it("Correlation filter with SQL action", async function(): Promise<void> {
+    await subscriptionClient.addRule(
+      "CorrelationRuleWithAction",
+      {
+        userProperties: {
+          color: "red"
+        }
+      },
+      "SET priority = 'High'"
+    );
+
+    const rules = await subscriptionClient.getRules();
+    should.equal(rules.length, 1, "rule");
+    should.equal(rules[0].name, "CorrelationRuleWithAction");
+
+    await sendOrders();
+    const receivedMsgs = await receiveOrders(subscriptionClient);
+
```

Comment:
Can you add one more check to see whether the `action` actually updates the attribute ?